### PR TITLE
Added watch functionality to sparkl render

### PR DESCRIPTION
**Pull request by miklosduma**

### CODE REVIEW

File: sparkl_cli/cmd_render.py [hunk 1]
```diff
@@ -23,47 +29,226 @@ def parse_args(subparser):
...
+    Gets the last modification date of
+    the specified file.
+    """
+    # Get date of latest modification
+    stats = os.stat(src_file)
+    latest_mod = stats[8]
+    return latest_mod
+
+
+def html_generator(src_file, previous_change, rate):
```
Comment (jacobythwaites) [inline sparkl_cli/cmd_render.py]: Cd you rename `rate` to `interval` throughout?

File: sparkl_cli/cmd_render.py [hunk 2]
```diff
@@ -23,47 +29,226 @@ def parse_args(subparser):
...
+
+        # Use generator to update file.
+        if args.output:
+            write_on_change(args.output, watcher)
+            return
+
+        # Or just return generator.
+        return watcher
+
+    # You can only watch local files. (I.e. -w must be used with -f)
```
Comment (jacobythwaites) [inline sparkl_cli/cmd_render.py]: # The --watch flag must be combined with --file flag.

File: sparkl_cli/test/test_put_source_rm.py [hunk 3]
```diff
@@ -7,94 +7,142 @@
...
         assert(line == "<html>\n")
 
+    def test_render_monitor(self):
+        result = sparkl(
+            "render",
+            PRIMES_FILE_PATH,
+            file=True,
+            alias=TEST_ALIAS,
+            watch=True
+        )
```
Comment (jacobythwaites) [inline sparkl_cli/test/test_put_source_rm.py]: I'd plonk this close ) at the end of the line above actually.

File: sparkl_cli/test/test_put_source_rm.py [hunk 4]
```diff
@@ -7,94 +7,142 @@
...
+        copyfile(PRIMES_FILE_PATH, temp_path)
+
+        # Start monitoring in a separate process
+        monitor_process = Process(target=sparkl,
+                                  args=("render", temp_path),
+                                  kwargs={"alias": TEST_ALIAS,
+                                          "watch": True,
+                                          "file": True,
+                                          "output": "tmp.html",
+                                          "rate": 1})
```
Comment (jacobythwaites) [inline sparkl_cli/test/test_put_source_rm.py]: ... like here!

**Comment by miklosduma** [comment]
@jacobythwaites see #b638061 for fixes

**Pull request merged**